src/neo/Ledger/TransactionRouter.cs

```diff
@@ -16,12 +19,8 @@ public TransactionRouter(NeoSystem system)
 
         protected override void OnReceive(object message)
         {
-            if (message is not Transaction tx) return;
-            system.Blockchain.Tell(new Blockchain.PreverifyCompleted
-            {
-                Transaction = tx,
-                Result = tx.VerifyStateIndependent(system.Settings)
-            }, Sender);
+            if (message is not Preverify preverify) return;
+            system.Blockchain.Tell(new PreverifyCompleted(preverify.Transaction, preverify.Relay, preverify.Transaction.VerifyStateIndependent(system.Settings)), Sender);
```

Comment:
How we will know that the message it's our reverification, in order to not do it again?

Reply:
What do you mean? This is `Preverify`, not `Reverify`...

Reply:
I mean, the answer is the same for verify and reverify, so we cannot distinguish asynchronously.

Reply:
Why do you need to distinguish it? We need it in dBFT plugin to verify the payloads in `RecoveryMessage`.